coderd/taskname/taskname.go

```diff
@@ -67,6 +69,31 @@ func GetAnthropicModelFromEnv() anthropic.Model {
 	return anthropic.Model(os.Getenv("ANTHROPIC_MODEL"))
 }
 
+// generateSuffix generates a random hex string between `100` and `fff`.
+func generateSuffix() string {
+	numMin := 0x100
+	numMax := 0x1000
+	//nolint:gosec // We don't need a cryptographically secure random number generator for generating a task name suffix.
+	num := rand.IntN(numMax-numMin) + numMin
+
+	return fmt.Sprintf("%x", num)
+}
+
+func GenerateFallback() string {
+	// We have a 32 character limit for the name.
+	// We have a 5 character prefix `task-`.
+	// We have a 4 character suffix `-fff`.
+	// This leaves us with 23 characters for the middle.
+	//
+	// Unfortunately, `namesgenerator.GetRandomName(0)` will
+	// generate names that are longer than 23 characters, so
+	// we just trim these down to length.
+	name := strings.ReplaceAll(namesgenerator.GetRandomName(0), "_", "-")
+	name = name[:min(len(name), 23)]
```

Comment:
We could end up with `name = "something-"` here, resulting in `task-something--fff`. We need to trim the trailing `-`.